lib/dockerregistry/inventory.go

```diff
@@ -373,29 +374,57 @@ func (riid *RegInvImageDigest) PrettyValue() string {
 	return b.String()
 }
 
-func getRegistryTagsFrom(req stream.ExternalRequest) (*google.Tags, Errors) {
+func getRegistryTagsWrapper(req stream.ExternalRequest) (*google.Tags, error) {
+
+	var googleTags *google.Tags
+
+	var getRegistryTagsCondition wait.ConditionFunc = func() (bool, error) {
+		var err error
+
+		googleTags, err = getRegistryTagsFrom(req)
+
+		// We never return an error (err) in the second part of our return
+		// argument, because we don't want to prematurely stop the
+		// ExponentialBackoff() loop; we want it to continue looping until
+		// either we get a well-formed tags value, or until it hits
+		// ErrWaitTimeout. This is how ExponentialBackoff() uses the
+		// ConditionFunc type.
+		if err == nil && googleTags != nil {
+			return true, nil
+		}
+		return false, nil
+	}
+
+	err := wait.ExponentialBackoff(
+		stream.BackoffDefault,
+		getRegistryTagsCondition)
+
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+
+	return googleTags, nil
+}
+
+func getRegistryTagsFrom(req stream.ExternalRequest) (*google.Tags, error) {
 	defer req.StreamProducer.Close()
 
-	errors := make(Errors, 0)
 	reader, _, err := req.StreamProducer.Produce()
 	if err != nil {
-		errors = append(errors, Error{
-			Context: "processing request",
-			Error:   err})
-
+		klog.Warning("error reading from stream:", err)
 		// Skip google.Tags JSON parsing if there were errors reading from the
 		// HTTP stream.
-		return nil, errors
+		return nil, err
 	}
 
 	tags, err := extractRegistryTags(reader)
 	if err != nil {
-		errors = append(errors, Error{
-			Context: "parsing JSON",
-			Error:   err})
+		klog.Warning("error parsing *google.Tags from io.Reader handle:", err)
```

Comment:
Probably `klog.Warningf("error parsing tags: %v", err)` ; I don't think the types are important enough to be in the error message.